homeassistant/helpers/storage.py

```diff
@@ -173,15 +195,22 @@ def _async_cleanup_delay_listener(self):
     async def _async_callback_delayed_write(self, _now):
         """Handle a delayed write callback."""
         self._unsub_delay_listener = None
-        self._async_cleanup_stop_listener()
+        self._async_cleanup_final_write_listener()
         await self._async_handle_write_data()
 
-    async def _async_callback_stop_write(self, _event):
-        """Handle a write because Home Assistant is stopping."""
-        self._unsub_stop_listener = None
+    async def _async_callback_final_write(self, _event):
+        """Handle a write because Home Assistant is in final write state."""
+        self._unsub_final_write_listener = None
         self._async_cleanup_delay_listener()
         await self._async_handle_write_data()
 
+    async def _async_callback_stop(self, _event):
```

Comment:
Why do this? Couldn't we better inside the `_async_callback_delayed_write` check if we're stopping, and if so schedule a final write call ?

Reply:
🤦‍♂